discord/tool.gpt

```diff
@@ -0,0 +1,144 @@
+
+---
+Name: List Channels
+Description: List all text channels in all Discord guilds the bot has access to. Returns the name, ID, guild name, and guild ID for each channel.
+Tools: github.com/gptscript-ai/datasets/filter
+Credential: ./credential
+Share Context: Discord Context
+
+#!/usr/bin/env npm --silent --prefix ${GPTSCRIPT_TOOL_DIR} run tool -- listChannels
+
+---
+Name: Search Channels
+Description: Search for text channels in Discord guilds the bot has access to
+Tools: github.com/gptscript-ai/datasets/filter
+Credential: ./credential
+Share Context: Discord Context
+Param: query: the search query
+
+#!/usr/bin/env npm --silent --prefix ${GPTSCRIPT_TOOL_DIR} run tool -- searchChannels
+
+---
+Name: Get Channel History
+Description: Get the chat history for a text channel in Discord
+Tools: github.com/gptscript-ai/datasets/filter
+Credential: ./credential
+Share Tools: List Channels, Search Channels
+Share Context: Discord Context
+Param: channelid: the ID of the channel to get the history for
+Param: guildid: the ID of the guild that the channel belongs to
+Param: limit: the number of messages to return - recommend starting with 10 and increasing if necessary
+
+#!/usr/bin/env npm --silent --prefix ${GPTSCRIPT_TOOL_DIR} run tool -- getChannelHistory
+
+---
+Name: Get Channel History by Time
+Description: Get the chat history for a text channel in Discord within a specific time range
+Tools: github.com/gptscript-ai/datasets/filter
+Credential: ./credential
+Share Tools: List Channels, Search Channels
+Share Context: Discord Context
+Param: channelid: the ID of the channel to get the history for
+Param: guildid: the ID of the guild that the channel belongs to
+Param: limit: the maximum number of messages to return - recommend starting with 10 and increasing if necessary
+Param: start: the start time in RFC 3339 format
+Param: end: the end time in RFC 3339 format
+
+#!/usr/bin/env npm --silent --prefix ${GPTSCRIPT_TOOL_DIR} run tool -- getChannelHistoryByTime
+
+---
+Name: Get Thread History
+Description: Get the chat history for a particular thread in a Discord channel
+Tools: github.com/gptscript-ai/datasets/filter
+Credential: ./credential
+Share Tools: List Channels, Get Channel History
+Share Context: Discord Context
+Param: channelid: the ID of the channel containing the thread
+Param: guildid: the ID of the guild that the channel belongs to
+Param: threadid: the ID of the thread to get the history for
+Param: limit: the number of messages to return - recommend starting with 10 and increasing if necessary
+
+#!/usr/bin/env npm --silent --prefix ${GPTSCRIPT_TOOL_DIR} run tool -- getThreadHistory
+
+---
+Name: Search Messages
+Description: Search for messages across all accessible Discord channels. Returns messages matching the query, including thread information if the message is part of a thread.
```

Comment:
I haven't been able to get this to return messages from threads.

Reply:
honestly it's a bad tool anyway. I meant to leave a comment about it but forgot to submit it on the PR. I'm just gonna remove it